app/src/main/java/com/patrickgatewood/weather/injection/ApplicationModule.java

```diff
@@ -63,7 +64,7 @@ DarkSkyApi provideDarkskyApi(Retrofit retrofit) {
 
     @Provides
     @Singleton
-    WeatherViewModel provideWeatherViewModel(DarkSkyApi darkSkyApi) {
-        return new WeatherViewModel(darkSkyApi);
+    WeatherPresenter provideWeatherPresenter(DarkSkyApi darkSkyApi) {
```

Comment:
@nishtahir Ah, I see. So when the view is destroyed, the presenter should be ready for garbage collection and if the view is revisited, a new presenter should be instantiated. As for view state, I plan on writing some state utility methods in the view's interface rather than keeping track of the view state in the presenter.